### PR TITLE
Fix bug with multiple minibatch variables

### PR DESCRIPTION
**Pull request by ricardoV94**
## Description

Reported in https://discourse.pymc.io/t/verifying-that-minibatch-is-actually-randomly-sampling/14308

The bug occurred due to separate calls to model.logp (from model.datalogp and model.varlogp) that create distinct clones of the RandomIntegersRV underlying minibatch slicing. `compile_pymc` would not set any updates in this case

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc--7408.org.readthedocs.build/en/7408/

### CODE REVIEW

File: tests/variational/test_inference.py [hunk 1]
```diff
@@ -472,3 +472,25 @@ def test_fit_data_coords(hierarchical_model, hierarchical_model_data):
...
+        weights = pm.Normal("weights", 0, 10, shape=10)
+        pm.Normal(
+            "y",
+            mu=minibatch_feature @ weights,
+            sigma=0.01,
+            observed=minibatch_y,
+            total_size=len(y),
+        )
+        mean_field = pm.fit(10_000, obj_optimizer=pm.adam(learning_rate=0.01), progressbar=False)
+    np.testing.assert_allclose(mean_field.mean.get_value(), true_weights, rtol=1e-1)
```
Comment (jessegrabowski) [inline tests/variational/test_inference.py:496]: Does this test need to run the whole model and check parameter recovery? It should be enough to compile the function and check that `minibatch_feature` and `minibatch_y` change after each loss function execution right?

**Review by jessegrabowski — APPROVED** [review]

File: tests/variational/test_inference.py (hunk 1 above)
Comment (ricardoV94) [reply tests/variational/test_inference.py:496]: This should run pretty fast, with minibatch of 1. I think it's a useful integration test, we didn't have any linear regression minibatch test in the codebase.

Besides VI has a very complex logic leading to building the function that I rather treat as a black box
